### PR TITLE
[MLv2] Make long display name available.

### LINKED ISSUE #30857: [MLv2] Make foreign column display names consistent
If a column is coming from a foreign table (as a result of implicit or explicit join), its display name should look like:

```
{{ table_name }} → {{ column_name }}
```

Examples (for a query started from the sample orders table):

* Product → Category
* People → Email

Note: other columns should just use their display name

[Related Slack thread](https://metaboat.slack.com/archives/C01LQQ2UW03/p1684432822052399)

### PR DESCRIPTION
**Pull request by snoe**
Fixes #30857 
Export `display-name` function and adds `long-display-name` to `display-info` results.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30882)

### CODE REVIEW

File: src/metabase/lib/js.cljs [hunk 1]
```diff
@@ -113,6 +113,16 @@
        (update :table update-keys u/->camelCaseEn)
        (clj->js :keyword-fn u/qualified-name))))
 
+(defn ^:export display-name
+  "Given an opaque Cljs object, return a plain JS object with a name useful for display.
+  See `:metabase.lib.metadata.calculation/display-name` ."
+  ([a-query x]
+   (display-name a-query -1 x))
+  ([a-query stage-number x]
+   (display-name a-query stage-number x "default"))
```
Comment (metamben) [inline src/metabase/lib/js.cljs:122]:
```suggestion
   (display-name a-query stage-number x :default))
```

File: src/metabase/lib/js.cljs [hunk 2]
```diff
@@ -113,6 +113,16 @@
...
+  ([a-query stage-number x style]
+   (lib.core/display-name a-query stage-number x (keyword style))))
```
Comment (metamben) [inline src/metabase/lib/js.cljs:124]: Isn't this missing the conversion of the result to JS stuff?

File: src/metabase/lib/metadata/calculation.cljc [hunk 3]
```diff
@@ -255,6 +255,7 @@
 (mr/register! ::display-info
   [:map
    [:display-name :string]
+   [:long-display-name {:optional true} :string]
```
Comment (metamben) [inline src/metabase/lib/metadata/calculation.cljc:258]: 👍 (I've forgotten this when I implemented aggregation operators.)

**Review by metamben — CHANGES_REQUESTED** [review]
Either remove `display-name` (I understand the FE doesn't need it now) or convert the results to JS.

File: src/metabase/lib/js.cljs (hunk 2 above)
Comment (snoe) [reply src/metabase/lib/js.cljs]: Since it's returning a string its (iiuc) unnecessary. But I remove since it's unwanted.

**Review by metamben — APPROVED** [review]

**Review by kulyk — APPROVED** [review]

**Pull request merged**